### PR TITLE
Optimise page header first steps

### PR DESCRIPTION
**Pull request by sndrs**
- batches first steps DOM reads/write
- moves all feature detection scope class insertion to before the CSS, so there's no need to re-calc the render tree

These show the time between beginning parsing the HTML and first redraw triggered by the inline CSS being parsed. 

before:
screen shot 2015-06-26 at 15 10 30

after:
screen shot 2015-06-26 at 15 13 16

### CODE REVIEW

**Comment by janua** [comment]
This looks good. Can we see the affect of this in ophan?

File: common/app/views/fragments/stylesheetsUse.scala.html [hunk 1]
```diff
@@ -52,15 +52,6 @@
 
             // GO!
             useCss();
-
```
Comment (johnduffell) [inline common/app/views/fragments/stylesheetsUse.scala.html]: good spot - this probably should have been associated with a switch so it did get remembered

**Comment by sndrs** [comment]
> Can we see the affect of this in ophan?

possibly. if it has an effect though, it's likely to be in responsiveness *after* first paint, so possibly not. this is more of a 'this is a better idea' following the chat at google yesterday.

**Comment by uplne** [comment]
Really nice. The effect is small so I would be surprised if there will be something noticeable in the ophan but at this point every small improvement adds to the overall performance. So go Alex :+1:

**Pull request merged**